app/Admin/src/Http/Controllers/Auth/AuthController.php

```diff
@@ -0,0 +1,59 @@
+<?php
+
+namespace App\Admin\Http\Controllers\Auth;
+
+use App\Admin\Http\Requests\Auth\LoginRequest;
+use App\Admin\Http\Requests\Auth\LogoutRequest;
+use App\Admin\Http\Resources\Auth\AdminResource;
+use App\Models\User;
+use App\Platform\Enums\RoleName;
+use Illuminate\Http\Response;
+use Illuminate\Support\Facades\Auth;
+use Illuminate\Validation\ValidationException;
+use Knuckles\Scribe\Attributes\Authenticated;
+use Knuckles\Scribe\Attributes\BodyParam;
+use Knuckles\Scribe\Attributes\Endpoint;
+
+class AuthController
+{
+    /**
+     * @throws ValidationException
+     */
+    #[Endpoint('Admin/Login', <<<'DESC'
+    This endpoint enable users with admin role to sign in
+ DESC)]
+    #[BodyParam('email', 'password', required: true)]
+    public function login(LoginRequest $request): AdminResource
+    {
+        $credentials = $request->only('email', 'password');
+
+        $user = User::where('email', $request->email)->first();
+
+        if (!$user || !$user->hasRole(RoleName::ADMIN)) {
+            throw ValidationException::withMessages([
+                'error' => ['Invalid credentials'],
+            ])->status(403);
+        }
+
+        if (Auth::attempt($credentials)) {
```

Comment:
You can negate this if check as well and append it to the if check above with ||. It will clean up the throwing of the exception at the bottom of the method and the happy path will be outside of the if statements. 

This is called defensive programming.

Reply:
Additionally, you can use the default message that Laravel provides for the exception message.